lighthouse-core/lib/dependency-graph/base-node.js

```diff
@@ -168,96 +168,79 @@ class BaseNode {
 
   /**
    * Clones the entire graph connected to this node filtered by the optional predicate. If a node is
-   * included by the predicate, all nodes along the paths between the two will be included. If the
-   * node that was called clone is not included in the resulting filtered graph, the method will throw.
+   * included by the predicate, all nodes along the paths between the node and the root will be included. If the
+   * node this was called on is not included in the resulting filtered graph, the method will throw.
    * @param {function(Node):boolean} [predicate]
    * @return {Node}
    */
   cloneWithRelationships(predicate) {
     const rootNode = this.getRootNode();
 
-    /** @type {function(Node): boolean} */
-    let shouldIncludeNode = () => true;
-    if (predicate) {
-      const idsToInclude = new Set();
-      rootNode.traverse(node => {
-        if (predicate(node)) {
-          node.traverse(
-            node => idsToInclude.add(node.id),
-            node => node._dependencies.filter(parent => !idsToInclude.has(parent))
```

Comment:
dangers of `Set<any>`...we were checking the `parent` node object against a set of string IDs, so weren't saving any time here

Reply:
nice fix!!

Reply:
I guess this one is difficult to test since it's just a marginal time savings huh

Reply:
> I guess this one is difficult to test since it's just a marginal time savings huh

yeah, and even at its worst it's really not that bad compared to other things we do with the graph